nil pointer deref in spot market request resource

The state refresh func for spot market request doesn't check for err, and sometimes make the TF provider panic. 

```
------- Stderr: -------
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0xf75e56]

goroutine 185 [running]:
github.com/terraform-providers/terraform-provider-packet/packet.resourceStateRefreshFunc.func1(0xc000217f50, 0xc000217d98, 0x2, 0x1, 0x1, 0x0)
	/opt/teamcity-agent/work/daa78b1b053319a8/src/github.com/terraform-providers/terraform-provider-packet/packet/resource_packet_spot_market_request.go:300 +0x246
github.com/terraform-providers/terraform-provider-packet/vendor/github.com/hashicorp/terraform-plugin-sdk/helper/resource.(*StateChangeConf).WaitForState.func1(0xc0004a14a0, 0xc0006f1110, 0xc000030e00, 0xc00003f9e0, 0xc0006f8890, 0xc0006f8888)
```	

Code causing this in `resource_packet_spot_market_request.go`:

```
        for _, d := range smr.Devices {

            dev, _, _ := client.Devices.Get(d.ID, nil)   // <-------------------
            if dev.State != "active" {
                break
            } else {
                finished = true
            }
        }


```

@jasmingacic do you mind takin a look? This was your code.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -8,6 +8,7 @@ IMPROVEMENTS:
 
 BUG FIXES:
 - [#180](https://github.com/terraform-providers/terraform-provider-packet/pull/159) Fixes device import 
+- [#183](https://github.com/terraform-providers/terraform-provider-packet/pull/184) Added missing error handling when waiting for spot instances to complete
 
 ## 2.5.0 (October 18, 2019)
 

--- a/packet/resource_packet_spot_market_request.go
+++ b/packet/resource_packet_spot_market_request.go
@@ -1,6 +1,7 @@
 package packet
 
 import (
+	"fmt"
 	"time"
 
 	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
@@ -289,7 +290,7 @@ func resourceStateRefreshFunc(d *schema.ResourceData, meta interface{}) resource
 		smr, _, err := client.SpotMarketRequests.Get(d.Id(), &packngo.GetOptions{Includes: []string{"project", "devices", "facilities"}})
 
 		if err != nil {
-			return nil, "", err
+			return nil, "", fmt.Errorf("Failed to fetch Spot market request with following error: %s", err.Error())
 
 		}
 		var finished bool
@@ -298,7 +299,7 @@ func resourceStateRefreshFunc(d *schema.ResourceData, meta interface{}) resource
 
 			dev, _, err := client.Devices.Get(d.ID, nil)
 			if err != nil {
-				return nil, "failed", err
+				return nil, "", fmt.Errorf("Failed to fetch Device with following error: %s", err.Error())
 			}
 			if dev.State != "active" {
 				break

--- a/packet/resource_packet_spot_market_request.go
+++ b/packet/resource_packet_spot_market_request.go
@@ -296,7 +296,10 @@ func resourceStateRefreshFunc(d *schema.ResourceData, meta interface{}) resource
 
 		for _, d := range smr.Devices {
 
-			dev, _, _ := client.Devices.Get(d.ID, nil)
+			dev, _, err := client.Devices.Get(d.ID, nil)
+			if err != nil {
+				return nil, "failed", err
+			}
 			if dev.State != "active" {
 				break
 			} else {

